cmeutils/structure.py

```diff
@@ -8,6 +8,106 @@
 from cmeutils.geometry import get_plane_normal, angle_between_vectors
 
 
+def angle_distribution(
+        gsd_file, A_name, B_name, C_name, start=0, stop=-1
+):
+    """Returns the bond angle distribution for a given triplet of particles 
+    
+    Parameters
+    ----------
+    gsdfile : str
+        Filename of the GSD trajectory.
+    A_name, B_name, C_name : str
+        Name(s) of particles that form the angle triplet 
+        (found in gsd.hoomd.Snapshot.particles.types)
+        They must be given in the same order as they form the angle
+    start : int
+        Starting frame index for accumulating bond lengths.
+        Negative numbers index from the end. (default 0)
+    stop : int
+        Final frame index for accumulating bond lengths. (default -1)
+
+    Returns
+    -------
+    1-D nump.array 
+        Array of bond angles in degrees
+
+    """
+    angles = []
+    trajectory = gsd.hoomd.open(gsd_file, mode="rb")
+    angle_name = "-".join([A_name, B_name, C_name])
+    for snap in trajectory[start: stop]:
+        if angle_name not in snap.angles.types:
+            raise ValueError(
+                    f"Angle {angle_name} not found in snap.angles.types. "
+                    "A_name, B_name, C_name must match the order "
+                    "as they appear in snap.angles.types."
+                )
+        for idx, angle_id in enumerate(snap.angles.typeid):
+            _angle_name = snap.angles.types[angle_id]
+            if _angle_name == angle_name:
+                pos1 = snap.particles.position[snap.angles.group[idx][0]]
+                img1 = snap.particles.image[snap.angles.group[idx][0]]
+                pos2 = snap.particles.position[snap.angles.group[idx][1]]
+                img2 = snap.particles.image[snap.angles.group[idx][1]]
+                pos3 = snap.particles.position[snap.angles.group[idx][2]]
+                img3 = snap.particles.image[snap.angles.group[idx][2]]
+                pos1_unwrap = pos1 + (img1 * snap.configuration.box[:3])
+                pos2_unwrap = pos2 + (img2 * snap.configuration.box[:3])
+                pos3_unwrap = pos3 + (img3 * snap.configuration.box[:3])
```

Comment:
you could use freud unwrap if you wanted to support non-orthogonal boxes, but it's not urgent

Reply:
yeah, I thought about that as well.  I say we can add that later if the need arises.